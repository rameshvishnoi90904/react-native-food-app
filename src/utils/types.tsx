export type FoodItem = {
    id: number,
    label: string,
    description: string,
    displayOrder: number,
    unitPriceFractional: number,
    currency: string,
    imageUrl: string,
    itemStock: {
        quantityLeft: number
    }
}

export type InnerSectionList = {
    sectionId: string,
    sectionDescription: string,
    disabledReason: string,
    list: FoodItem[],
    subSection: SectionItem[]
}

export type SectionItem = {
    id: number,
    label: string,
    description: string,
    displayOrder: number,
    disabled: Boolean,
    disabledReason: string|null,
    data: InnerSectionList[],
    outerSectionTitle: string
}


export type NavigationScreenPropsType = {
    Home: undefined;
    Detail: {item: FoodItem, disabledSection: boolean};
};

export interface CartState {
    [key: number]: CartItem;
}

export type CartItem = {
    foodId: number,
    quantity: number
}

export type dispatchActionType = {
    type: string;
    foodId: number;
}
