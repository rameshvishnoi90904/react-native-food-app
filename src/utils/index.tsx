import { FoodItem, SectionItem } from "./types";
/**
 * 
 * @param item section item
 * @param sectionTitle outer nested section title
 * @returns section item with outer nested section title
 */
const formatInnerSection = (item: any, sectionTitle: string) => {
    const foodItems = [...item.items];
    foodItems.sort((a: FoodItem, b: FoodItem) => (a.displayOrder - b.displayOrder));

    const toReturn: SectionItem = {
        id: item.id, 
        label: item.label,
        outerSectionTitle: sectionTitle,
        description: item.description,
        disabled: item.disabled,
        disabledReason: item.disabledReason,
        displayOrder: item.displayOrder,
        data: [{list: item.items, sectionId: item.id, sectionDescription: item.description, disabledReason: item.disabled ? item.disabledReason : "", subSection: []}],
    };
    return toReturn;
}
/**
 * 
 * @param item section data from api
 * @returns section item compatible with react native SectionList component
 */
export const formatSection = (item: any) : SectionItem => {
    const foodItems = [...item.items];
    const subSections = [...item.subSections];

    foodItems.sort((a: FoodItem, b: FoodItem) => (a.displayOrder - b.displayOrder));
    subSections.sort((a: SectionItem, b: SectionItem) => (a.displayOrder - b.displayOrder));
    
    const formattedSubSection: SectionItem[] = subSections.map((sItem) => formatInnerSection(sItem, item.label))

    const toReturn = {
        id: item.id, 
        label: item.label,
        description: item.description,
        disabled: item.disabled,
        outerSectionTitle: "",
        disabledReason: item.disabledReason,
        displayOrder: item.displayOrder,
        data: [{list: item.items, sectionId: item.id, sectionDescription: item.description, disabledReason: item.disabled ? item.disabledReason : "", subSection: formattedSubSection}],
    };
    return toReturn;
}