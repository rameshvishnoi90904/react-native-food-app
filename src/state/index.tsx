
import { createContext } from 'react';
import { dispatchActionType, CartState, SectionItem } from '../utils/types';

export type AppContextType = {
    state: {cart: CartState, foodSection: SectionItem[]};
    dispatch: (action: dispatchActionType) => void;
}

export const AppContext = createContext<AppContextType | null>(null);