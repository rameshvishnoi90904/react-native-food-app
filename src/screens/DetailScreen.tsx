import React, { useContext, useEffect, useState } from "react"
import { Image, StyleSheet, View, Text, TouchableOpacity, Dimensions, ScrollView } from "react-native"
import type { NativeStackScreenProps } from '@react-navigation/native-stack';
import { CartItem, FoodItem, NavigationScreenPropsType } from "../utils/types";
import { SafeAreaView, useSafeAreaFrame } from "react-native-safe-area-context";
type Props = NativeStackScreenProps<NavigationScreenPropsType, 'Detail'>;
import RenderHtml from 'react-native-render-html';
import { AppContext, AppContextType } from "../state";

const DetailScreen = ({route}: Props) => {
    const item: FoodItem = route.params.item;
    const disabledSection: boolean = route.params.disabledSection;
    const {state, dispatch} = useContext(AppContext) as AppContextType
    const cart = state.cart;
    
    const currentCart = cart[item.id] || {};

    const addToCart = () => {
        dispatch({type: 'add', foodId: item.id})
    }
    const removeFromCart = () => {
        dispatch({type: 'remove', foodId: item.id})
    }

    let actionButton, priceValue;
    if (item.itemStock.quantityLeft > 0) {
        if (currentCart.quantity > 0) {
            const totalPrice = item.unitPriceFractional * currentCart.quantity/100;
            actionButton = (
                actionButton = (
                    <View style={[styles.cardFunctions, {marginRight: 4}]}>
                        <TouchableOpacity testID="remove-cart" onPress={() => removeFromCart()}>
                            <Text style={styles.cartAction}>-</Text>
                        </TouchableOpacity>
                        <Text style={styles.cartQuantity}>{currentCart.quantity}</Text>
                        <TouchableOpacity testID="add-cart" onPress={() => addToCart()}>
                            <Text style={styles.cartAction}>+</Text>
                        </TouchableOpacity>
                    </View>
                )
            )
            priceValue = (
                <TouchableOpacity style={[styles.addButton, {marginLeft: 4}]} onPress={addToCart}>
                    <Text style={styles.addText}>Update ($ {totalPrice})</Text>
                </TouchableOpacity>
            )
        } else if (disabledSection) {
            actionButton = (
                <TouchableOpacity style={styles.addButton} disabled>
                    <Text style={styles.addText}>Not Available</Text>
                </TouchableOpacity>
            )
        } else {
            actionButton = (
                <TouchableOpacity testID="add-cart-button" style={styles.addButton} onPress={addToCart}>
                    <Text style={styles.addText}>Add</Text>
                </TouchableOpacity>
            )
        }
       
    } else {
        actionButton = (
            <TouchableOpacity style={[styles.addButton, styles.soldOutButton]} disabled>
                <Text style={styles.addText}>Sold out</Text>
            </TouchableOpacity>
        )
    }

    return (
        <View style={styles.container}>
            <ScrollView style={{flex: 1,}} contentContainerStyle={{}}>
                <Image src={item.imageUrl} style={styles.imagePreview} resizeMode="contain"/>
                <View style={styles.textContainer}>
                    <Text style={styles.header} testID="item-label">{item.label}</Text>
                    {
                        item.description ?
                        <RenderHtml
                            contentWidth={width}
                            source={{html: item.description}}
                        />
                        :
                        <Text style={styles.description}>Description not available</Text>
                    }
                </View>
            </ScrollView>
            <View style={styles.buttonWrapper}>
                {actionButton}
                {priceValue}
            </View>
        </View>


    )
}
const {width} = Dimensions.get("screen")
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'rgb(255, 254, 250)',
    },
    imagePreview: {
        width: width,
        height: width * 3/4,
    },
    textContainer: {
        flex: 1, 
        paddingHorizontal: 20,
    },
    header: {
        fontSize: 20,
        marginVertical: 10
    },
    description: {
        lineHeight: 21,
    },
    addButton: {
        backgroundColor: 'rgb(250, 85, 83)',
        alignItems: 'center',
        height: 40,
        justifyContent: 'center',
        borderRadius: 4,
        flex: 1
    },
    soldOutButton: {
        backgroundColor: '#fdcece',
    },
    addText: {
        color: 'white'
    },
    cardFunctions: {
        borderWidth: 1,
        flexDirection: 'row' ,
        height: 40, 
        alignItems: 'center',
        borderRadius: 4,
         flex: 1,
     },
     cartAction: {
         fontSize: 18,
         padding: 10
     },
     cartQuantity: {
         flex: 1,
         fontSize: 20,
         textAlign: 'center'
     },
     buttonWrapper: {
        marginHorizontal: 20,
        flexDirection: 'row'
     }
})
export default DetailScreen