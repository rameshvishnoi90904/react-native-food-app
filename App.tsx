/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React, { useEffect, useState } from 'react';
import {
  SafeAreaView,
} from 'react-native';
import MenuScreen from './src/screens/MenuScreen';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import {CartState, NavigationScreenPropsType, SectionItem, dispatchActionType} from './src/utils/types';
import DetailScreen from './src/screens/DetailScreen';
import { AppContext } from './src/state';
import { fetchMenu } from './src/api';
const Stack = createNativeStackNavigator<NavigationScreenPropsType>();

function App(): React.JSX.Element {
  const [cart, updateCart] = useState<CartState>({})
  const [foodSection, setFoodSection] = useState<SectionItem[]>([])

  useEffect(() => {
    fetchMenu().then((value: SectionItem[]) => {
      setFoodSection(value)
  })
  },[])

  const dispatch = (action: dispatchActionType) => {
    const toReturn = {...cart}

    switch (action.type) {
      case 'add': 
        if (toReturn[action.foodId]) {
          toReturn[action.foodId].quantity++
        } else {
          toReturn[action.foodId] = {
            quantity: 1,
            foodId: action.foodId
          }
        }
      break;
      case 'remove': 
        toReturn[action.foodId].quantity--
      break;
    }
    updateCart(toReturn)
  }
  const state = {cart: cart, foodSection: foodSection}
  return (
    <SafeAreaView style={{flex: 1}}>
      <AppContext.Provider value={{state, dispatch}}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="Home" component={MenuScreen} />
            <Stack.Screen name="Detail" component={DetailScreen} />
          </Stack.Navigator>
        </NavigationContainer>
      </AppContext.Provider>
    </SafeAreaView>
  );
}

export default App;
