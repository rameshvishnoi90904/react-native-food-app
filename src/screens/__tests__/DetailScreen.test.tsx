import 'react-native';
import React from 'react';
import { screen, render, act, fireEvent, waitFor } from '@testing-library/react-native';
import {it, jest, expect} from '@jest/globals';
import { NavigationContainer } from '@react-navigation/native';
import { RootNavigator } from '../../../App';
import { AppContext } from '../../state';
import DetailScreen from '../DetailScreen';
import mockData from '../../api/fakeData.json'
import { SectionItem, NavigationScreenPropsType, CartState } from '../../utils/types';
import { formatSection } from '../../utils';

import { createNativeStackNavigator } from '@react-navigation/native-stack';

const Stack = createNativeStackNavigator<NavigationScreenPropsType>();

const mockSections  = mockData.sections.map((item: any) : SectionItem => formatSection(item));
mockSections.sort((a: SectionItem, b: SectionItem) => (a.displayOrder - b.displayOrder))

it('render DetailScreen with mock data', () => {
    const dispatch = jest.fn();
    const state = {
        cart: {},
        foodSection: [...mockSections]
    }

    const firstFoodItem = mockSections[0].data[0].list[0];
    render(
        <AppContext.Provider value={{state, dispatch}}>
            <NavigationContainer>
                <Stack.Navigator>
                    <Stack.Screen name="Detail" component={DetailScreen} initialParams={{item: firstFoodItem, disabledSection: false}}/>
                </Stack.Navigator>
            </NavigationContainer>
        </AppContext.Provider>
    ).toJSON();
    expect(screen).toMatchSnapshot();
})



it('render DetailScreen with cart interaction', async () => {
    const dispatch = jest.fn();

    const firstFoodItem = mockSections[0].data[0].list[0];
    const cart: CartState = {};
    cart[firstFoodItem.id] = {
        foodId: firstFoodItem.id,
        quantity: 1
    }

    const state = {
        cart: cart,
        foodSection: [...mockSections]
    }

    render(
        <AppContext.Provider value={{state, dispatch}}>
            <NavigationContainer>
                <Stack.Navigator>
                    <Stack.Screen name="Detail" component={DetailScreen} initialParams={{item: firstFoodItem, disabledSection: false}}/>
                </Stack.Navigator>
            </NavigationContainer>
        </AppContext.Provider>
    )

    const incrementButton = await screen.findByTestId("add-cart");

    act(() => {
        fireEvent.press(incrementButton);
    })

    const decrementButton = await screen.findByTestId("remove-cart");

    act(() => {
        fireEvent.press(decrementButton);
    })

    expect(dispatch).toBeCalledTimes(2)
})

it("render DetailScreen with disabled section", () => {
    const dispatch = jest.fn();
    const state = {
        cart: {},
        foodSection: [...mockSections]
    }

    const firstFoodItem = mockSections[0].data[0].list[0];
    render(
        <AppContext.Provider value={{state, dispatch}}>
            <NavigationContainer>
                <Stack.Navigator>
                    <Stack.Screen name="Detail" component={DetailScreen} initialParams={{item: firstFoodItem, disabledSection: true}}/>
                </Stack.Navigator>
            </NavigationContainer>
        </AppContext.Provider>
    );
    expect(screen).toMatchSnapshot();
})


it("render DetailScreen with soldout state section", () => {
    const dispatch = jest.fn();
    const state = {
        cart: {},
        foodSection: [...mockSections]
    }

    const firstFoodItem = mockSections[0].data[0].list[0];
    const soldOutItem = {
        ...firstFoodItem,
        description: '',
        itemStock: {
            quantityLeft: 0
        }
    }
    render(
        <AppContext.Provider value={{state, dispatch}}>
            <NavigationContainer>
                <Stack.Navigator>
                    <Stack.Screen name="Detail" component={DetailScreen} initialParams={{item: soldOutItem, disabledSection: false}}/>
                </Stack.Navigator>
            </NavigationContainer>
        </AppContext.Provider>
    );
    expect(screen).toMatchSnapshot();
})