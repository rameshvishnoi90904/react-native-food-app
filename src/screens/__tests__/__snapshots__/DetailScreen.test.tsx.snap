// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render DetailScreen with disabled section 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "rgb(255, 254, 250)",
                "flex": 1,
              }
            }
          >
            <RCTScrollView
              contentContainerStyle={{}}
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View>
                <Image
                  resizeMode="contain"
                  src="https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt"
                  style={
                    {
                      "height": 562.5,
                      "width": 750,
                    }
                  }
                />
                <View
                  style={
                    {
                      "flex": 1,
                      "paddingHorizontal": 20,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "fontSize": 20,
                        "marginVertical": 10,
                      }
                    }
                    testID="item-label"
                  >
                    Rasa Sayang Burg
                  </Text>
                  <View
                    style={
                      [
                        {},
                        undefined,
                        undefined,
                      ]
                    }
                    testID="html"
                  >
                    <View
                      style={
                        [
                          {},
                          undefined,
                          undefined,
                        ]
                      }
                      testID="body"
                    >
                      <Text
                        allowFontScaling={true}
                        selectable={false}
                        style={
                          [
                            {
                              "fontSize": 14,
                            },
                            undefined,
                            undefined,
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={true}
                          selectable={false}
                          style={
                            [
                              {
                                "fontSize": 14,
                              },
                              undefined,
                              undefined,
                            ]
                          }
                        >
                          Marrying great taste and sustainability, the 100% plant-based burger features Quorn's meat-free burger patty packed with the goodness of mycoprotein - a nutritious protein source that is high in protein and fibre, and low in saturated fat.
                        </Text>
                        <Text
                          testID="br"
                        >
                          

                        </Text>
                        <Text
                          testID="br"
                        >
                          

                        </Text>
                        <Text
                          allowFontScaling={true}
                          selectable={false}
                          style={
                            [
                              {
                                "fontSize": 14,
                              },
                              undefined,
                              undefined,
                            ]
                          }
                        >
                          Paired with a special, vegan sunny-side up egg, the burger is topped with fresh cucumber slices, a generous dollop of kampong sambal sauce, sweet caramelized onions, creamy vegan aioli, all sitting on a bed of crunchy lettuce and sandwiched between a fresh wholemeal bun.
                        </Text>
                        <Text
                          testID="br"
                        >
                          

                        </Text>
                        <Text
                          testID="br"
                        >
                          

                        </Text>
                        <Text
                          allowFontScaling={true}
                          selectable={false}
                          style={
                            [
                              {
                                "fontSize": 14,
                              },
                              undefined,
                              undefined,
                            ]
                          }
                        >
                          *Patty is NOT allium-free. You can opt for allium-free patty. 
                        </Text>
                        <Text
                          testID="br"
                        >
                          

                        </Text>
                        <Text
                          allowFontScaling={true}
                          selectable={false}
                          style={
                            [
                              {
                                "fontSize": 14,
                              },
                              undefined,
                              undefined,
                            ]
                          }
                        >
                          *Option to remove onions.
                        </Text>
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
            <View
              style={
                {
                  "flexDirection": "row",
                  "marginHorizontal": 20,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "rgb(250, 85, 83)",
                    "borderRadius": 4,
                    "flex": 1,
                    "height": 40,
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                    }
                  }
                >
                  Not Available
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Detail"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;

exports[`render DetailScreen with mock data 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "rgb(255, 254, 250)",
                "flex": 1,
              }
            }
          >
            <RCTScrollView
              contentContainerStyle={{}}
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View>
                <Image
                  resizeMode="contain"
                  src="https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt"
                  style={
                    {
                      "height": 562.5,
                      "width": 750,
                    }
                  }
                />
                <View
                  style={
                    {
                      "flex": 1,
                      "paddingHorizontal": 20,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "fontSize": 20,
                        "marginVertical": 10,
                      }
                    }
                    testID="item-label"
                  >
                    Rasa Sayang Burg
                  </Text>
                  <View
                    style={
                      [
                        {},
                        undefined,
                        undefined,
                      ]
                    }
                    testID="html"
                  >
                    <View
                      style={
                        [
                          {},
                          undefined,
                          undefined,
                        ]
                      }
                      testID="body"
                    >
                      <Text
                        allowFontScaling={true}
                        selectable={false}
                        style={
                          [
                            {
                              "fontSize": 14,
                            },
                            undefined,
                            undefined,
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={true}
                          selectable={false}
                          style={
                            [
                              {
                                "fontSize": 14,
                              },
                              undefined,
                              undefined,
                            ]
                          }
                        >
                          Marrying great taste and sustainability, the 100% plant-based burger features Quorn's meat-free burger patty packed with the goodness of mycoprotein - a nutritious protein source that is high in protein and fibre, and low in saturated fat.
                        </Text>
                        <Text
                          testID="br"
                        >
                          

                        </Text>
                        <Text
                          testID="br"
                        >
                          

                        </Text>
                        <Text
                          allowFontScaling={true}
                          selectable={false}
                          style={
                            [
                              {
                                "fontSize": 14,
                              },
                              undefined,
                              undefined,
                            ]
                          }
                        >
                          Paired with a special, vegan sunny-side up egg, the burger is topped with fresh cucumber slices, a generous dollop of kampong sambal sauce, sweet caramelized onions, creamy vegan aioli, all sitting on a bed of crunchy lettuce and sandwiched between a fresh wholemeal bun.
                        </Text>
                        <Text
                          testID="br"
                        >
                          

                        </Text>
                        <Text
                          testID="br"
                        >
                          

                        </Text>
                        <Text
                          allowFontScaling={true}
                          selectable={false}
                          style={
                            [
                              {
                                "fontSize": 14,
                              },
                              undefined,
                              undefined,
                            ]
                          }
                        >
                          *Patty is NOT allium-free. You can opt for allium-free patty. 
                        </Text>
                        <Text
                          testID="br"
                        >
                          

                        </Text>
                        <Text
                          allowFontScaling={true}
                          selectable={false}
                          style={
                            [
                              {
                                "fontSize": 14,
                              },
                              undefined,
                              undefined,
                            ]
                          }
                        >
                          *Option to remove onions.
                        </Text>
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
            <View
              style={
                {
                  "flexDirection": "row",
                  "marginHorizontal": 20,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "rgb(250, 85, 83)",
                    "borderRadius": 4,
                    "flex": 1,
                    "height": 40,
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
                testID="add-cart-button"
              >
                <Text
                  style={
                    {
                      "color": "white",
                    }
                  }
                >
                  Add
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Detail"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;

exports[`render DetailScreen with soldout state section 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "rgb(255, 254, 250)",
                "flex": 1,
              }
            }
          >
            <RCTScrollView
              contentContainerStyle={{}}
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View>
                <Image
                  resizeMode="contain"
                  src="https://storage.googleapis.com/kaluga-core-active-storage/9uvvn2dj2x6y28uqkw7e1gs9bnkt"
                  style={
                    {
                      "height": 562.5,
                      "width": 750,
                    }
                  }
                />
                <View
                  style={
                    {
                      "flex": 1,
                      "paddingHorizontal": 20,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "fontSize": 20,
                        "marginVertical": 10,
                      }
                    }
                    testID="item-label"
                  >
                    Rasa Sayang Burg
                  </Text>
                  <Text
                    style={
                      {
                        "lineHeight": 21,
                      }
                    }
                  >
                    Description not available
                  </Text>
                </View>
              </View>
            </RCTScrollView>
            <View
              style={
                {
                  "flexDirection": "row",
                  "marginHorizontal": 20,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#fdcece",
                    "borderRadius": 4,
                    "flex": 1,
                    "height": 40,
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                    }
                  }
                >
                  Sold out
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Detail"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
